### ISSUE TITLE
formToModifier not being called on a update form

### ISSUE THREAD

**Issue by ripdog** [aldeed/meteor-autoform#859]
I have a form which I repurpose for both insert and update of appointment items. I have seperate "time" and "date" items in the schema for appointments, the latter of which is hidden from the form and set with the time, which is a string. (The day is set outside of the form).

This logic was contained in docToForm, so 2.1 broke my form updating. No problem, move the logic into a function and call that from both `docToForm` and `docToModifier`, right? I tried that, and now my update field calls neither `docToForm` or `docToModifier`.

Schema: 

```javascript
	date: {
		type: Date,
		label: "Date",
		index: true,
		custom: function() {
			return checkDate(this, true);
		}
	},
	time: {
		type: String,
		label: "Time",
 		regEx: /^[0-2]?\d:\d\d [APap]m|M$/
	},
```

Form: 
```
<template name="insertAppointmentForm">
	{{#autoForm collection=appointmentList doc=currentDoc id="insertAppointmentFormInner" type=currentType}}
...
		{{> afQuickField name='time' value=timePreset type='timePicker'}}
...
		{{> afFieldInput name='date' value=sessionDate type="hidden"}}
	{{/autoForm}}
</template>
```
I have verified that the form is correctly marked as update via `AutoForm.getCurrentDataForForm("insertAppointmentFormInner")`. Validation is default `submitThenKeyup`.

Form JS: 
```javascript
AutoForm.hooks({
	insertAppointmentFormInner: {
...
		formToDoc: function(doc) {
			prepareDocForDB(doc)
		},
		formToModifier: function(doc) {
			console.log(doc);
			prepareDocForDB(doc)
		},
...
   }
});

prepareDocForDB = function(doc) {

	if (typeof doc.time === "string") {
		var datestring = moment(Session.get("date")).format("YYYY-MM-DD ") + doc.time;
		doc.date = moment(datestring, "YYYY-MM-DD hh:mm A").utc().toDate();
	}
	return doc;
}
```

I have set breakpoints on both of the `prepareDocForDB` calls on both Firefox and Chrome, never got called during an update operation.

Thanks so much for your time and hard work! :)

**Comment by fpoirier1** [comment]
I am facing the same issue using the `method-update` form type.

**Comment by gwendall** [comment]
+1

**Comment by pajooh** [comment]
+1

**Comment by yim1990** [comment]
+1

```
Hooks = {
  form: {},
  global: {
    before: {},
    after: {},
    formToDoc: [],
    docToForm: [],
    onSubmit: [],
    onSuccess: [],
    onError: [],
    beginSubmit: [],
    endSubmit: []
  }
};
```

I think this is the problem. there is no "formToModifier" hook.
This feature should have been tested.

**Comment by apavia3** [comment]
+1

**Comment by haydnhkim** [comment]
+1

**Comment by trance4mation** [comment]
+1

**Comment by jethro91** [comment]
+1

**Comment by kabalin** [comment]
The case for me as well, `formToModifier` is not called for "method-update" form type. The workaround is to amend modifier in `before` hook, e.g.:
```
...
   editLocationForm: {
     before: {
       "method-update": function(modifier) {
         var latlng = modifier.$set.location.split(',');
         doc.$set.location = { type: "Point", coordinates: [ parseFloat(latlng[1]), parseFloat(latlng[0]) ] };
         return modifier;
       } 
     },
```

**Comment by laurentpayot** [comment]
Same bug here with a quickForm update. Thanks @kabalin for the workaround.

**Comment by aramk** [comment]
@laurentpayot this was fixed with #1058

**Comment by aldeed** [comment]
I haven't released yet but hopefully soon. Need to get some things working in devel branch.

**Comment by pcuci** [comment]
+1 

Same bug for basic `type="update"` form, formToDoc hook not called, skipped altogether

**Comment by pcuci** [comment]
@aldeed - reproduction repo here: https://github.com/pcuci/autoform-practice

There another issue!

`formToDoc` gets called on `type="update"` forms instead of `formToModifier` - but only for the text field (`tags` field in the repo)

The `range` input doesn't trigger `formToDoc` nor `formToModifier` on update.

**Comment by fabiokandrade** [comment]
+1

**Comment by structuresound** [comment]
+1 @ 5.3.2

**Comment by wizonesolutions** [comment]
@aldeed is 5.3.2 coming soon? `meteor add ` won't let me install anything un-semver-like, and there don't seem to be any published forks including this fix (not that I think using a fork is a very good idea).

For now... @kabalin's workaround worked well. I thought it wouldn't because `before` is supposedly run after the form is deemed valid. I guess the form being valid and the values being valid against the schema are two different things, and it clearly runs before validation of the underlying schema happens. My spin on it was:

````javascript
before: {
  update: function (modifier) {
    return CommonHooks.formToModifier(modifier);
  }
}   
````

**Comment by shkomg** [comment]
+1
having the same bug as @pcuci 

formToDoc not fired on update.

**Comment by aldeed** [comment]
Pretty sure this was fixed a few releases ago.

**Comment by maksimkurb** [comment]
@aldeed, I have the same bug as @pcuci 
My form:
```jade
+quickForm(id="editSchool" collection="School" type="update" doc=editingDoc fields="name, address, phone, workBegin, workEnd, description" buttonContent=false)
```

My hooks:
```js
AutoForm.hooks
  editSchool:
    formToDoc: (doc) ->
      console.log doc
      if doc.workBegin
        time = doc.workBegin.split ':'
        date = new Date()
        date.setHours time[0], time[1]
        doc.workBegin = date
      if doc.workEnd
        time = doc.workEnd.split ':'
        date = new Date()
        date.setHours time[0], time[1]
        doc.workEnd = date
      return doc
```

If I remove `type="update"` from quickform, `formToDoc` hook will trigger as expected.
